src/drivers/intel/cavs/ssp.c

```diff
@@ -328,14 +327,16 @@ static inline int ssp_set_config(struct dai *dai,
 	sscr2 |= (ssp->params.quirks & SOF_DAI_INTEL_SSP_QUIRK_PSPSRWFDFD) ?
 		SSCR2_PSPSRWFDFD : 0;
 
-#if defined(CONFIG_ICELAKE)
-	if (!config->ssp.mclk_rate || config->ssp.mclk_rate > F_38400_kHz) {
+	if (!config->ssp.mclk_rate ||
+	    config->ssp.mclk_rate > ssp_freq[ARRAY_SIZE(ssp_freq) - 1].freq) {
```

Comment:
maybe use a define for the max index, e.g. MAX_SSP_FREQ_INDEX to avoid those -1, this is error-prone.

Reply:
I've made define.